xc/device_common.ml

```diff
@@ -334,15 +334,20 @@ let is_upstream_qemu domid =
 		with_xs (fun xs -> xs.Xs.read (Printf.sprintf "/libxl/%d/dm-version" domid)) = "qemu_xen"
 	with _ -> false
 
-let qmp_write domid cmd =
-	let open Qmp_protocol in
-	let c = connect (Printf.sprintf "/var/run/xen/qmp-libxl-%d" domid) in
-	finally
-		(fun() ->
-			try
-				negotiate c;
-				write c cmd
-			with e ->
-				error "Caught exception attempting to write qmp message: %s" (Printexc.to_string e);
-		)
-		(fun() -> close c)
+let qmp_write_and_read domid ?(read_result=false) cmd  =
```

Comment:
In this way you can avoid setting the read result in the qmp_write_and_read and the behaviour is the one you would expect